### DISCUSSION TITLE
Please, why this refreshInterval Option is not working for me?

### DISCUSSION THREAD

**Discussion by spavlovic77** [vercel/swr#1626] [General] [answered]
const fetchSupplyInfo = (slice, accounts) => (...args) => {
          const vSupply =  slice.methods.sliceParams().call({ from: accounts[0] }).then((data) => {return data[0]})
          return vSupply
      }
      const { data: vSupply, error2 } = useSWR('supplyInfo', {fetcher: fetchSupplyInfo(slice, accounts)}, { refreshInterval: 1000 })
          console.log('vSupply', vSupply)

Revalidation on focus works just fine. But autorefresh does not work.

**Comment by huozhi** [comment, marked as answer]
The fetcher you passed looks like an expression call instead of a function? and the `error2` should be `error` I think.

**Reply by spavlovic77** [reply]
the fetcher calls web3 method to interact with Smart Contract on Blockchain. I can succesfully fetch data. The only think which is not working is refreshInterval.

Any idea why?

**Reply by huozhi** [reply]
The code snippet provided above is kind of hard to understand, do you have a codesandbox link or a minimal repro demo for it?

**Reply by spavlovic77** [reply]
Solved. You inspired me. Below is the working hook. Thanks a lot.      


const fetchSupplyInfo = (slice, accounts) => {
          const vSupply =  slice.methods.sliceParams().call({ from: accounts[0] }).then((data) => {return data[0]})
          return vSupply
      }
      const { data: vSupply, error2 } = useSWR([slice, accounts],fetchSupplyInfo, { refreshInterval: 1000 })
          console.log('vSupply', vSupply)
